server/container/util/src/test/java/reactor/retry/RetryTestUtils.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2017 Pivotal Software Inc, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.retry;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import org.reactivestreams.Publisher;
+
+import reactor.core.publisher.Flux;
+
+public class RetryTestUtils {
+
+	static void assertDelays(Queue<? extends IterationContext<?>> retries, Long... delayMs) {
+		assertEquals(delayMs.length, retries.size());
+		int index = 0;
+		for (Iterator<? extends IterationContext<?>> it = retries.iterator(); it.hasNext(); ) {
+			IterationContext<?> repeatContext = it.next();
+			assertEquals(delayMs[index].longValue(), repeatContext.backoff().toMillis());
+			index++;
+		}
+	}
+
+	static void assertRandomDelays(Queue<? extends IterationContext<?>> retries, int firstMs, int maxMs) {
+		long prevMs = 0;
+		int randomValues = 0;
+		for (IterationContext<?> context : retries) {
+			long backoffMs = context.backoff().toMillis();
+			assertTrue("Unexpected delay " + backoffMs, backoffMs >= firstMs && backoffMs <= maxMs);
+			if (backoffMs != firstMs && backoffMs != prevMs)
```

Comment:
curly brackets

Reply:
I don't expect to change these classes as explained https://github.com/linagora/james-project/pull/3099#discussion_r377637758